ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -433,7 +434,7 @@ private void onNextImpl(AppendEntriesReplyProto reply) {
         case INCONSISTENCY:
           grpcServerMetrics.onRequestInconsistency(getFollowerId().toString());
           LOG.warn("{}: received {} reply with nextIndex {}", this, reply.getResult(), reply.getNextIndex());
-          updateNextIndex(reply.getNextIndex());
+          updateNextIndex(Math.max(getFollower().getMatchIndex() + 1, reply.getNextIndex()));
```

Comment:
If a follower shutdowns, clear the logs and meta conf, and then restarts (just as what we do in `runTestBootstrapReconf`), the leader will still maintains the matchIndex (previously before shutdown) of this follower.  The changes here will cause the nextIndex never being correctly updated. I think we can tolerate some inconsistent AppendEntries in trade for fault tolerance and robustness. What do you think @szetszwo ?

Reply:
> ... clears the logs and meta conf, ...

The follower cannot clear the existing log.  Otherwise, the committed entries can be removed.  As an example, see the description of RATIS-1677.   BTW, `runTestBootstrapReconf` is starting new peers but not removing existing logs.

> ... Match index in FollowerInfoImpl is only allowed to be changed monotonously.  ...

Once the log entries are matched, it cannot be changed to unmatched later on.  This is an assumption of the Raft algorithm.